File: src/test/java/spring/practice/springMicroservice/feignPractice/OuterClassTest.java
package spring.practice.springMicroservice.feignPractice;

import org.junit.jupiter.api.Test;

/**
 * Created by kimchanjung on 2020-04-25 2:51 오후
 */
class OuterClassTest {

    @Test
    public void test() {
        OuterClass.InnerClass innerClass = new OuterClass().new InnerClass();
        innerClass.showOuterName();

        OuterClassStaticInnerClass.StaticInnerClass.showOuterName();
    }
}

File: src/test/java/spring/practice/kotlin/basicSyntaxPractice/OuterClassTest.kt
package spring.practice.kotlin.basicSyntaxPractice

import org.junit.jupiter.api.Test

/**
 * Created by kimchanjung on 2020-04-25 2:18 오후
 */
internal class OuterClassTest {

    @Test
    fun 중첩_클래스가_정상적으로_동작한다() {
        // Given
        val staticNestedClass = OuterClass.StaticNestedClass()
        val innerClass = OuterClass().InnerClass()

        // When
        staticNestedClass.getOuterName()
        innerClass.getOuterName()
    }
}